README.md

```diff
@@ -32,3 +28,152 @@
+* [Create and publish modules to npm](https://quickleft.com/blog/creating-and-publishing-a-node-js-module/)
+
+### Globals, Process and Buffer###
+As we have learnt, node's module system discourages the use of globals; however node provides a few important globals for use to utilize. Here we list the most important ones.
+
+- [Buffer](http://nodejs.org/api/buffer.html): To handle binary data, node provides us with the global Buffer object. Buffer instances represent memory allocated independently of V8's heap. [Here](http://www.w3resource.com/node.js/nodejs-buffer.php) is an example of how to use buffers.	 
+- [Global](http://nodejs.org/api/globals.html): This module represent the global scope. It is not a good practice to use this global. When you define a global variable in a browser, its scope is the global one, but in node is different: you define a variable inside a module without "var" and its scope is the module itself.
+- [Process](http://nodejs.org/api/process.html): Process is a node module that can be accessed anywhere without the need to be required. It exposes process manipulation such as signalling, exiting, the process id (pid), and more.
+## Asynchronous Patterns ##
+### Event loop ###
+- [Understanding Node Event Loop](https://nodesource.com/blog/understanding-the-nodejs-event-loop)
+- [Why Single threaded and Event Loop](http://blog.mixu.net/2011/02/01/understanding-the-node-js-event-loop/)
+
+### Async functions ###
+- [Callback Hell](http://callbackhell.com/)
+- [Using async functions in practice](http://www.sebastianseilund.com/nodejs-async-in-practice)
+- [Async iteration patterns](http://metaduck.com/01-asynchronous-iteration-patterns.html)
+
+### Control Flow and Exception Handling ###
+- [Solutions to control the flow](http://book.mixu.net/node/ch7.html)
+- [Managing the asynchronous nature of NodeJS](http://code.tutsplus.com/tutorials/managing-the-asynchronous-nature-of-nodejs--net-36183)
+- [The Async library](https://github.com/caolan/async)
+- [Error Handling in NodeJS](https://www.joyent.com/developers/node/design/errors)
+
+## Communication ##
+### EventEmitter ###
+Many objects in Node emit events: a net.Server emits an event each time a peer connects to it, a fs.readStream emits an event when the file is opened. All objects which emit events are instances of events.EventEmitter.
+[http://nodejs.org/api/events.html](http://nodejs.org/api/events.html)
+
+### Cluster ###
+A single instance of Node runs in a single thread. To take advantage of multi-core systems the user will sometimes want to launch a cluster of Node processes to handle the load. [http://nodejs.org/api/cluster.html](http://nodejs.org/api/cluster.html)
+
+### Streams ###
+A stream is an abstract interface implemented by various objects in Node. Streams are pipes that let you easily read data from a source and pipe it to a destination. [http://nodejs.org/api/stream.html](http://nodejs.org/api/stream.html): 
+
+## RESTful API ##
```

Comment:
This section introduces RESTful principles, along with the HTTP protocol basics.